src/Constraints.elm

```diff
@@ -0,0 +1,400 @@
+                                    ]
+                                )
+                            |> Maybe.withDefault []
+                            |> List.append [ todoCommentCheck comment ]
+                            |> List.filterMap identity
+                            |> List.append notExistingArgsCheck
+                            |> List.append notAnnotatedArgsCheck
+
+                    _ ->
+                        let
+                            firstLine =
+                                String.split "\n" comment
+                                    |> List.head
+                        in
+                        firstLine
+                            |> Maybe.map
+                                (\line ->
+                                    [ startingCapitalizedCheck line
+                                    , startingVerbCheck line
+                                    , endingPeriodCheck line
+                                    , startingSpaceCheck line
+                                    ]
+                                )
+                            |> Maybe.withDefault []
+                            |> List.append [ todoCommentCheck comment ]
+                            |> List.filterMap identity
+
+            Nothing ->
+                case entity.eType of
+                    Models.Function params ->
+                        params
+                            |> List.map NotAnnotatedArgument
+                            |> (::) NoEntityComment
+
+                    Models.Record fields ->
+                        fields
+                            |> List.map NotAnnotatedArgument
+                            |> (::) NoEntityComment
+
+                    _ ->
+                        [ NoEntityComment ]
+        )
+            |> filterIgnored ignored
+
+
+getViolationsDangling : Models.Comment -> List Type -> List Type
+getViolationsDangling ( _, comment ) ignored =
+    let
+        commentLine =
+            if String.startsWith "--" comment then
+                comment
+
+            else
+                String.split "\n" comment
+                    |> List.head
+                    |> Maybe.withDefault ""
+    in
+    [ startingCapitalizedCheck commentLine
+    , startingSpaceCheck commentLine
+    , endingPeriodCheck commentLine
+    , commentTypeCheck commentLine
+    , todoCommentCheck commentLine
+    ]
+        |> List.filterMap identity
+        |> filterIgnored ignored
+
+
+getViolationsTopLevel : Maybe Models.Comment -> List Type -> List Type
+getViolationsTopLevel maybeComment ignored =
+    (case maybeComment of
+        Just ( _, comment ) ->
+            [ startingSpaceCheck comment
+            , todoCommentCheck comment
+            ]
+                |> List.filterMap identity
+
+        Nothing ->
+            [ NoTopLevelComment ]
+    )
+        |> filterIgnored ignored
+
+
+startingCapitalizedCheck : String -> Maybe Type
```

Comment:
Could you maybe move all "*Check" functions to a module called "checks" (or something similar) and remove "Check" suffix?